### PR TITLE
Configurable signals

### LINKED ISSUE #521: Make the kill signal configurable
Namely:

https://github.com/remy/nodemon/blob/dd6ad0d879a1520554410cd662fac982c06f4c7c/lib/monitor/run.js#L195

I see there has been some windows-related issues, but I happen to be using a framework which uses SIGUSR2 for something else. As a matter of fact, if I use the https://github.com/bnoordhuis/node-heapdump module, nodemon will simply trigger heap dumps to be produced.

In my case at least, I'd like to be able to use nodemon.json to tell nodemon to SIGTERM instead of SIGUSR2. Any reasons why this would not be desirable?

### PR DESCRIPTION
**Pull request by stelcheck**
Fixes #521 

Todo:

- [ ] Unit tests
- [ ] Documentation

About unit tests, any recommendation on how I should approach it, where I should put the test?

### CODE REVIEW

File: lib/monitor/run.js [hunk 1]
```diff
@@ -264,7 +264,7 @@ function kill(child, signal, callback) {
       // spawning processes like `coffee` under the `--debug` flag, it'll spawn
       // it's own child, and that can't be killed by nodemon, so psTree gives us
       //  an array of PIDs that have spawned under nodemon, and we send each the
-      //   SIGUSR2 signal, which fixes #335
+      //   configured signal (defaul: SIGUSR2) signal, which fixes #335
```
Comment (ronkorving) [inline lib/monitor/run.js]: *default

**Comment by remy** [comment]
Good to see work being done on this, can you make sure to follow the contributing guidelines, specifically on the commit messages so that I can just merge and it'll auto-deploy (otherwise I have to rebase and mess around re-writing the commits). Looks good so far!

**Comment by stelcheck** [comment]
Wah, completely missed your last message! Sorry about that.

Oh, no problem. What do you expect in terms of documentation and UT? Any advices/clues/insults?

**Comment by Pchelolo** [comment]
+1 on this. 

@stelcheck Are you going to continue on this? If not, I could take over the PR as I really wanna see this done :)

**Comment by stelcheck** [comment]
@pchelolo, no objections.

**Comment by andrewshawcare** [comment]
Any updates on this?

Using nodemon in docker for non-node-related projects, having SIGTERM would be great.

**Comment by nmccready** [comment]
This is an excellent, highly desirable PR!

**Comment by nmccready** [comment]
@remy ?

**Comment by krevels** [comment]
would love to have this for a java process that I am monitoring

**Comment by remy** [comment]
Probably wants some docs in the readme too…

**Comment by nmccready** [comment]
Thanks @remy

**Comment by remy** [comment]
But I've merged, and once tests finish, it'll auto deploy a new version (I think 1.11.0).

**Pull request merged**